girder/molecules/server/openbabel.py

```diff
@@ -55,6 +55,12 @@ def cjson_to_ob_molecule(cjson):
     return mol
 
 def autodetect_bonds(cjson):
+    # Only autodetect bonds if we have 3D coordinates
+    if ('atoms' not in cjson or 'coords' not in cjson['atoms'] or
+        '3d' not in cjson['atoms']['coords'] or
+        len(cjson['atoms']['coords']['3d']) == 0):
```

Comment:
Lets use jsonpath to make the check a little more readable, something like:

`parse('atoms.coords.3d').find(cjson)`

The are lots of examples, in the repo ...

Reply:
Okay.